Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -1548,13 +1593,32 @@ PhysicalParticleContainer::AddPlasmaFlux (PlasmaInjector const& plasma_injector,
                 const long ip = poffset[index] + i_part;
                 pa_idcpu[ip] = amrex::SetParticleIDandCPU(pid+ip, cpuid);
 
-                // This assumes the flux_pos is of type InjectorPositionRandomPlane
-                const XDim3 r = (fine_overlap_box.ok() && fine_overlap_box.contains(iv)) ?
-                  // In the refined injection region: use refinement ratio `rrfac`
-                  flux_pos->getPositionUnitBox(i_part, rrfac, engine) :
-                  // Otherwise: use 1 as the refinement ratio
-                  flux_pos->getPositionUnitBox(i_part, amrex::IntVect::TheUnitVector(), engine);
-                auto pos = getCellCoords(overlap_corner, dx, r, iv);
+                // Determine the position of the particle within the cell
+                XDim3 pos;
+                XDim3 r;
+#ifdef AMREX_USE_EB
+                if (inject_from_eb) {
+#if defined(WARPX_DIM_3D)
+                    pos.x = overlap_corner[0] + (iv[0] + 0.5_rt + eb_bnd_cent_arr(i,j,k,0))*dx[0];
```

Comment:
Is this putting all of the particles at the same location, the EB boundary center? Is there a way to distribute the particles around on the EB surface?

Reply:
Yes, that's absolutely correct: all the particles are emitted from a single point. (See also the PR description.) 
We should indeed distribute the particles over the EB surface. This should be doable but will take a bit more time. This will be added in a follow-up PR.

Reply:
I should have read the whole PR description :)